src/AutoBuild.jl

```diff
@@ -160,8 +164,57 @@ function build_tarballs(ARGS, src_name, src_version, sources, script,
         error("Cannot register with a local deployment!")
     end
 
+    universe_path = nothing
+    if universe
+        if deploy
+            if deploy_repo != "universe"
+                error("When deploying from a universe, must use --deploy=universe")
+            end
+            deploy_universe = true
+            deploy_bin_repo = deploy_jll_repo = "JuliaBinaryWrappers/$(src_name)_jll.jl"
+        end
+
+        # Find the universe directory by ascending from pwd() until we find a "universes" directory
```

Comment:
I thought about it, but I do kinda want it explicitly available in Yggdrasil in case you need to go in and edit something. Putting it in Scratch makes it really hard to find.

Reply:
It's also the same thing that `git` does to find its `.git` folders.

Reply:
I'd like to have a way for these universes to get GC'ed.  I think Fredrik and I are going to add optional user-defined time limits to scratch spaces so that you can e.g. `@get_scratch("Universe-$(universe)"; lifetime=Days(3))` and if you don't use it again in three days, it gets auto-cleaned up.

That's the main reason I want to use scratch spaces.  Perhaps we can add a symlink in `build`, so that if you do `julia build_tarballs.jl --universe=SS_rebuild --deploy=universe`, it then creates not only `build/SuiteSparse*.tar.gz`, but also `build/universes/SS_rebuild` which points to the universe where that was built?

Reply:
Why should universes get explicitly gc'ed? Seems like something that should be very much up to the user.

Reply:
I think these should be auto-GC'ed because the usecase for universes is for experimentation, not for users to use this as their main source of JLLs.  I want to use these within Yggdrasil to allow us to open a PR that builds multiple, interdependent packages, and use the builds from the bottom of the stack as dependencies for the top of the stack, without having to register in-between.  If we're creating a universe per-PR, I want to be sure that they will be removed after a while.